src/main/java/xyz/vegaone/scribe/service/ScribeService.java

```diff
@@ -0,0 +1,40 @@
+package xyz.vegaone.scribe.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Slf4j
+public class ScribeService {
+
+    private FileDiscoveryServiceImpl fileDiscoveryServiceImpl;
+
+    private FileReaderServiceImpl fileReaderServiceImpl;
```

Comment:
I think it's a good practice for good encapsulation and for loose coupling of the code, to use the interface type in the field declaration, and not the implementation. @Callin , correct me if I'm wrong.

Reply:
mostly for loose coupling